### PR TITLE
fix(token-input): format max value with thousand separators

### LINKED ISSUE #295: Token input 'max' value not validated nor formatted
**Describe the bug**
When clicking the max button, the input is filled with raw data, without being formatted (that is, adding the commas for the thousand separators), nor it's validated. 

For the latter it's a bit weird, as it's the dapp defining the max value, it shouldn't be something to be checked, but it happens. I don't think it's necessary to fix that as it shouldn't pass a value greater than max (in the example video, the value has been hardcoded).

**To Reproduce**
Se the video

**Expected behavior**
Clicking max, should fill the input with the value properly formatted

**Screenshots**

https://github.com/user-attachments/assets/e33f3bfb-265e-4440-b650-bb6cf25f6e0b

### PR DESCRIPTION
**Pull request by gabitoesmiapodo**
## Summary

Closes #295

When clicking the Max button in `TokenInput`, the value bypassed `NumericFormat`'s formatting
pipeline because `BigNumberInput`'s `inputRef` was not connected to the `NumericFormat`
instance. A `displayValue` state now provides a controlled value string so `NumericFormat`
applies thousand separators on external value changes (max click, token change) as it does
during typing.

## Changes

- Add `displayValue` state to `BigNumberInput` with render-time sync from the `value` prop;
  maps `0n` to `''` to preserve the placeholder
- Update `updateValue` to set `prevValueRef` before `onChange` so user-typed intermediate
  strings (trailing dots, trailing zeros) are not overwritten by the sync
- Pass `value: displayValue` through `renderInputProps` so `NumericFormat` operates in
  controlled mode
- Clear `amountError` in `handleSetMax` to dismiss any prior validation error on max click
- Add 4 tests: empty placeholder on `0n`, thousand-separator formatting on external value
  change, reset to placeholder, user-typed "0" preserved

## Acceptance criteria

- [x] Clicking Max fills the input with the value formatted with thousand separators
- [x] Normal typing and validation continue to work as before
- [x] Placeholder shows when no amount is entered (`0n`)

### Automated tests

Modified: `src/components/sharedComponents/BigNumberInput.test.tsx`

```
pnpm test src/components/sharedComponents/BigNumberInput.test.tsx
```

### Manual verification

1. Open the token input demo
2. Connect a wallet with a token balance > 999
3. Click "Max" — value should appear with thousand separators (e.g. `1,234.56`)
4. Type a value manually — formatting and validation should work as before
5. Clear the input — placeholder `0.00` should reappear

## Breaking changes

None.

## Screenshots

None.

## Checklist

- [x] Self-reviewed my own diff
- [x] Tests added or updated
- [ ] Docs updated (if applicable)
- [x] No unrelated changes bundled in

### CODE REVIEW

File: src/components/sharedComponents/BigNumberInput.tsx [hunk 1]
```diff
@@ -68,8 +68,22 @@ export const BigNumberInput: FC<BigNumberInputProps> = ({
...
+  const prevDecimalsRef = useRef(decimals)
 
-  // update inputValue when value changes
+  // Sync displayValue when an external change updates value or decimals (e.g. max click, token change).
+  // Using render-time state update to avoid a visible flash between renders.
+  if (prevValueRef.current !== value || prevDecimalsRef.current !== decimals) {
+    prevValueRef.current = value
+    prevDecimalsRef.current = decimals
+    setDisplayValue(value === BigInt(0) ? '' : formatUnits(value, decimals))
+  }
```
[the remark this answers is not in the document]
Comment (gabitoesmiapodo) [reply src/components/sharedComponents/BigNumberInput.tsx]: Fixed in `fdbe05a`. Replaced `useState('')` with a lazy initializer that calls `formatUnits(value, decimals)` when `value !== 0n`, so first render reflects the prop.

[the remark this answers is not in the document]
Comment (gabitoesmiapodo) [reply src/components/sharedComponents/BigNumberInput.tsx:85]: This matches the React-documented "storing information from previous renders" pattern exactly. React recognizes the idiom: the component returns early and re-renders immediately, so no intermediate paint occurs. StrictMode double-invocation is safe because `prevValueRef` is updated before `setDisplayValue`, making the second invocation a no-op. A `useLayoutEffect` still incurs two renders before paint with no meaningful difference; a `useEffect` causes the visible flash the code comment explicitly calls out. The pattern is intentional.

File: src/components/sharedComponents/BigNumberInput.test.tsx [hunk 2]
```diff
@@ -124,3 +125,73 @@ describe('BigNumberInput', () => {
...
+    expect(input.value).toBe('')
+  })
+
+  it('formats value with thousand separators when value changes externally', async () => {
+    const { input, rerender } = renderWithNumericFormat()
+    rerender(parseUnits('1000', 18))
+    await waitFor(() => {
+      expect(input.value).toBe('1,000')
+    })
+  })
```
[the remark this answers is not in the document]
Comment (gabitoesmiapodo) [reply src/components/sharedComponents/BigNumberInput.test.tsx:185]: Fixed in `fdb5e09`. Added a test that mounts `BigNumberInput` with `parseUnits('1000', 18)` as the initial value and asserts the input displays `'1,000'`.

File: src/components/sharedComponents/BigNumberInput.test.tsx [hunk 3]
```diff
@@ -124,3 +125,78 @@ describe('BigNumberInput', () => {
...
+          value={value}
+          onChange={onChange}
+          renderInput={({ onChange: handleChange, value: displayVal, ...restProps }) => (
+            <NumericFormat
+              thousandSeparator
+              onValueChange={({ value: v }) => handleChange(v)}
+              value={displayVal as string | undefined}
+              // biome-ignore lint/suspicious/noExplicitAny: mirrors TokenAmountField pattern
+              {...(restProps as any)}
+            />
```
[the remark this answers is not in the document]
Comment (gabitoesmiapodo) [reply src/components/sharedComponents/BigNumberInput.test.tsx]: Fixed in 7bfa231. Added `inputRef: _inputRef` to the destructuring in the test helper so it no longer spreads into NumericFormat, matching the production pattern.

File: src/components/sharedComponents/BigNumberInput.tsx [hunk 4]
```diff
@@ -68,8 +68,24 @@ export const BigNumberInput: FC<BigNumberInputProps> = ({
...
+  )
+  const prevValueRef = useRef(value)
+  const prevDecimalsRef = useRef(decimals)
+
+  // Sync displayValue when an external change updates value or decimals (e.g. max click, token change).
+  // Using render-time state update to avoid a visible flash between renders.
+  if (prevValueRef.current !== value || prevDecimalsRef.current !== decimals) {
+    prevValueRef.current = value
+    prevDecimalsRef.current = decimals
+    setDisplayValue(value === BigInt(0) ? '' : formatUnits(value, decimals))
```
[the remark this answers is not in the document]
Comment (gabitoesmiapodo) [reply src/components/sharedComponents/BigNumberInput.tsx]: Fixed in 7821c8a. The render-time sync is now guarded with `if (renderInput)` so the native input path no longer triggers unnecessary re-renders.
